kv/memberlist/tcp_transport.go

```diff
@@ -147,6 +147,9 @@ func NewTCPTransport(config TCPTransportConfig, logger log.Logger, registerer pr
 	port := config.BindPort
 	for _, addr := range config.BindAddrs {
 		ip := net.ParseIP(addr)
```

Comment:
Could you please add a test case for the scenario where `addr` is not a valid IP address, to ensure we don't re-introduce this regression in the future?

Reply:
Done